tcbuilder/backend/deploy.py

```diff
@@ -354,6 +354,60 @@ def write_rootfs_to_raw_image(base_raw_img, output_raw_img, base_rootfs_partitio
                 args=(f"{dst_sysroot_dir}/{content}", "/"),
                 loading_msg=f"  Copying /{content}...")
 
+        # handle the combine of docker-storage bundle
+        bundle_dir = "bundle.tmp"
+        if os.path.exists(bundle_dir):
```

Comment:
Here we're just relying on the existence of a directory with a pre-determined name. Seems kind of fragile to me. Especially when we try to add support for raw images in the standalone `bundle` command, we may have to re-think or revise this.

Looking at what was done for the Tezi equivalent here. Internally to the build command we do `bundle` -> `combine`. But, looking at the `combine` command, it would need to be revised to add support for raw images if we want to have a similar workflow.

@lucas-akira Did you have any ideas/plans for raw-images and the `bundle`/`combine` workflow when you were adding initial support for raw-images?

Reply:
Well, the idea is to create the path anyway if it does not exist, and if it exists the code cleans it and recreates it. It follows the same way, or kind of, that the combine works. But instead of removing the folder at the end, it leaves it there. If you guys though that is better remove it I can do the same.

Reply:
> Looking at what was done for the Tezi equivalent here. Internally to the build command we do bundle -> combine. But, looking at the combine command, it would need to be revised to add support for raw images if we want to have a similar workflow.
>
> Did you have any ideas/plans for raw-images and the bundle/combine workflow when you were adding initial support for raw-images?

Actually the `combine` command already supports raw images. For an eventual `build` command support, the workflow that I was planning would be to internally call `bundle` followed by `combine`, just like it is right now with TEZI images.

That way we would avoid having part of the `bundle` command code doing essentially the same thing that `combine` already does.

Reply:
> Actually the combine command already supports raw images.

Oh okay so in that case this can just use the same workflow  as Tezi then?

Reply:
@jsrc27 @lucas-akira I updated the MR for the bundle bd12fd197118f0dde1c77a068e03a3083f1930d4 I'm following the same flow from the CLI command now, but with one exception. The compression of the .tar file does not make sense. For TEZI it makes sense, but not for raw images. So, I added some flags to tell the flow not to compress and decompress the file, which decreased the build processing time.

Take a look, thanks!

Reply:
> The compression of the .tar file does not make sense. For TEZI it makes sense, but not for raw images. So, I added some flags to tell the flow not to compress and decompress the file, which decreased the build processing time.

Right, given that during a `build` command the data of the bundled container images are tarred then soon after untarred into the raw image rootfs, I agree that it doesn't make sense to add the compression step to `docker-storage.tar` in this situation.

That being said, I think that the compression step should still be added by default when running the standalone `bundle` command, whether the container images are made for TEZI or raw images. I say this because in this case the resulting bundle directory will be saved on disk, and it could be stored to be reused for more than one OS image. And so it would be good if the contents are compressed to save disk space.

If doing the above then the `combine` command should be modified to be able to auto-detect if the Docker bundle tar file is compressed or not, and act accordingly.